### DISCUSSION TITLE
Система тормозит с количеством пользователей онлайн более 10.

### DISCUSSION THREAD

**Discussion by vldekov** [totumonline/totum-ru-issues-and-discussions#516] [Как это сделать?] [answered]
Столкнулись с проблемой когда пользователей онлайн стало более 10. В этом случае система начинает тормозить ( крутиться шестеренка более 1 минуты ). Как возможно решить эту проблему?

**Comment by vldekov** [comment]
Как снизить нагрузку на базу и файловую систему и ускорить работу сайта? Можно установить и настроить memcached.

Memcached – система кэширования данных в оперативной памяти. Так как оперативная памяти существенно быстрее дисковой, скорость работы с хранимыми в memcahed ресурсами значительно выше.

На услуге Хостинга memcached доступен отдельной опцией. Она подключается в настройках контейнера на вкладке «Информация». Объём хранилища вы определяете сами в диапазоне от 256 Мб до 2048 Мб.

На облачном VDS вы можете установить memcached-сервер через Терминал или SSH-доступ с помощью менеджера пакетов. А также настроить расширения PHP, Python или других языков для работы с ним.

Все инструкции – в [статье](https://www.netangels.ru/support/cloud-vds/memcached/)!

**Reply by vldekov** [reply]
Сделал кеширование сайта и индексирование таблиц. Пока вроде помогло.

**Comment by vldekov** [comment]
Кто то пробовал так сделать?

**Reply by erpic-ru** [reply]
Там дело не только в кэшировании, больше в оптимизации работы с таблицами: отключение отслеживания в нужных местах, разнесение на разные таблицы, оптимизация запросов, добавление индексов, асинхронные действия, докидывание ядер процессора, настройка postgres... да много всего :)

**Comment by ttm-alexeypolunin** [comment]
Как установлено. Это виртуалка (vds/vps)? Или это хостинг за 100 р./мес?

**Reply by vldekov** [reply]
Это VDS.

**Reply by ttm-alexeypolunin** [reply, marked as answer]
Увеличьте количество воркеров fpm: https://ru.docs.totum.online/ubuntu#pm

у вас должна быть оперативка на них. На 40 воркеров требуется ~2 гб.

Дальше смотрите что вы там у себя с архитектурой сделали, на предмет мест взять все/поделить на все/пересчитать все.

Если если таблица с заказами (циклы) и 10 чел одновремено работает в своих заказах, то табл циклов ставьте в Без отслеживания изменений.

Понятно, что если у вас на виртуалке 2 ядра и вы написали так, что таблица у одного пользователя открывается 10 секунд (не поставили разбивку итп) то у вас осталось одно ядро. Если это делают 2 человека, то у вас оба ядра заняты и все остальные ждут. Итп

**Reply by vldekov** [reply]
Есть информация про 5 версию тотум?
![image](https://github.com/totumonline/totum-ru-issues-and-discussions/assets/114401808/9f5bfe25-df38-4cb2-a5d2-68b820018ccf)

**Reply by ttm-alexeypolunin** [reply]
В работе инструкции по переходу.

**Reply by vldekov** [reply]
С 4 версии mit возможно будет перейти на 5 версию PRO ?

**Reply by ttm-alexeypolunin** [reply]
да

**Comment by vldekov** [comment]
Тотум умеет работать с кешированием ( memcached )?

**Reply by ttm-alexeypolunin** [reply]
Умеет, через execSSH, но так как вы спрашиваете это в этой ветке, то скорее всего вы не понимаете что такое memcaсhed, зачем применяется, граничные возможности итп.

Скорее всего вы думаете, что подключив его куда-то к серверу целиком, можно пофиксить ошибки в архитектуре. Те это типа такая «таблетка белая». Так не работает.

Но например на сервере демо мы пишем в мемкешед время исполнения всех скриптов всех схем, что бы не нагружать постгрес лишней тысячей транзакций в минуту и забираем оттуда эти данные раз в час.
